src/Database/Entities/Provider.php

```diff
@@ -12,17 +13,8 @@
  *
  * @SuppressWarnings(PHPMD.UnusedPrivateField) Suppress warning about "unused" $providerId.
  */
-class Provider
+class Provider extends Entity
```

Comment:
But why?

Reply:
@edwardmurrell-loyaltycorp because the EntityManager's `persist` method, which is used in the `ProviderController`, expects the provided object to extend or inherit `EntityInterface`.

Reply:
I'd like to see the restriction that EntityInterface can only be passed to our EntityManagerInterface, because if we're importing libraries that have their own entities we cant then use them without more work.

But for this PR I think that work is too large.